src/test/java/seedu/address/model/ModelManagerTest.java

```diff
@@ -209,9 +222,110 @@ public void updateMatchedTutor_modifyList_throwAssertionError() {
         TagsContainTagPredicate predicate = new TagsContainTagPredicate(ls);
         TypicalPersons.getTypicalTutors().stream().forEach(tutor -> modelManager.addTutor(tutor));
-        assertThrows(AssertionError.class, () -> modelManager.updateMatchedTutor(predicate, new ArrayList<>()));
+        assertThrows(AssertionError.class, () -> modelManager.updateMatchedTutor(predicate, new ArrayList<>(),
+                DANIEL));
+    }
+
+    @Test
+    public void clearMatchedTutorTest() {
+        // Expected empty match list
+        FilteredList<Tutor> expectedMatchList = new FilteredList<>(modelManager.getMatchedTutorList());
+
+        // Clear empty list
+        modelManager.clearMatchedTutor();
+        assertEquals(expectedMatchList, modelManager.getMatchedTutorList());
+
+        // Clear list after matching
+        ArrayList<Tag> ls = new ArrayList<>();
+        ls.addAll(DANIEL.getTags());
+        TagsContainTagPredicate predicate = new TagsContainTagPredicate(ls);
+        TypicalPersons.getTypicalTutors().stream().forEach(tutor -> modelManager.addTutor(tutor));
+        // Valid match as TypicalTutors have tutors that teach subject that student DANIEL wants
+        modelManager.updateMatchedTutor(predicate, ls, DANIEL);
+        modelManager.clearMatchedTutor();
+        assertEquals(expectedMatchList, modelManager.getMatchedTutorList());
     }
 
+    @Test
+    public void filterMatchedTutor_modifyList_success() {
+        ArrayList<Tag> ls = new ArrayList<>();
+        ls.addAll(DANIEL.getTags());
+        TagsContainTagPredicate predicate = new TagsContainTagPredicate(ls);
+        TypicalPersons.getTypicalTutors().stream().forEach(tutor -> modelManager.addTutor(tutor));
+        modelManager.addTutor(ENZIO);
+        CliTutors cliTutors = new CliTutors();
+        cliTutors.addTutor(ENZIO);
+        cliTutors.addTutor(ALICE);
+        FilteredList<Tutor> expectedTutorList = new FilteredList<>(cliTutors.getTutorList());
+        modelManager.updateMatchedTutor(predicate, ls, DANIEL);
+        assertEquals(expectedTutorList, modelManager.getMatchedTutorList());
+        // Check matched student
+        assertEquals(DANIEL, modelManager.getMatchedStudent());
+
+        // filter out females only
+        ArrayList<Gender> genderList = new ArrayList<>();
+        genderList.add(new Gender("F"));
+        GenderContainsGenderPredicate predicate1 = new GenderContainsGenderPredicate(genderList);
+        modelManager.filterMatchedTutor(predicate1);
+        cliTutors.removeTutor(ENZIO);
+        expectedTutorList = new FilteredList<>(cliTutors.getTutorList());
+        assertEquals(expectedTutorList, modelManager.getMatchedTutorList());
+        // Check matched student
+        assertEquals(DANIEL, modelManager.getMatchedStudent());
+
+        // filter out graduate qualification, no matches
+        ArrayList<Qualification> qualificationList = new ArrayList<>();
+        genderList.add(new Gender("F"));
```

Comment:
Perhaps you may try and kindly assist me in creating a femaleGender field instead of using `new Gender("F")`

But honestly doesnt really affect much